src/game_inventory.cpp

```diff
@@ -688,30 +688,50 @@ class comestible_inventory_preset : public inventory_selector_preset
         }
 
         bool sort_compare( const inventory_entry &lhs, const inventory_entry &rhs ) const override {
-            time_duration time_a = get_time_left( lhs.any_item() );
-            time_duration time_b = get_time_left( rhs.any_item() );
-            int order_a = get_order( lhs.any_item(), time_a );
-            int order_b = get_order( rhs.any_item(), time_b );
+            // Sort by comestibility first
+            const bool lhs_comestible = lhs.any_item()->is_comestible();
+            const bool rhs_comestible = rhs.any_item()->is_comestible();
+            if( lhs_comestible != rhs_comestible ) {
+                return !lhs_comestible && rhs_comestible;
+            }
+
+            // Then by rotten/fresh status depending on player's traits
+            const bool lhs_rotten = lhs.any_item()->rotten();
+            const bool rhs_rotten = rhs.any_item()->rotten();
+            if( lhs_rotten != rhs_rotten ) {
+                return prefer_rotten() ^ ( !lhs_rotten && rhs_rotten );
+            }
+
+            // Then by perishability
+            const bool lhs_imperishable = !lhs.any_item()->goes_bad();
+            const bool rhs_imperishable = !rhs.any_item()->goes_bad();
+            if( lhs_imperishable != rhs_imperishable ) {
+                return !lhs_imperishable && rhs_imperishable;
+            }
 
-            return order_a < order_b
-                   || ( order_a == order_b && time_a < time_b )
-                   || ( order_a == order_b && time_a == time_b &&
-                        inventory_selector_preset::sort_compare( lhs, rhs ) );
+            // Then by sealed status, skipping imperishable items
+            const bool lhs_sealed = is_sealed( lhs.any_item() );
+            const bool rhs_sealed = is_sealed( rhs.any_item() );
```

Comment:
Items can still rot in a sealed pocket if the pocket has a non-zero spoil multiplier (for example plastic bottles have a sealed spoil multiplier of 0.5). Addtionally unsealed pockets can also have a spoil multiplier less than 1 (for example condiment bottles have a unsealed spoil multiplier of 0.5).

I suggest to just compare the time left divided by the spoil multiplier (and use multiplication for numerical stability, i.e. `lhs_time_left * rhs_spoil_multiplier < rhs_time_left * lhs_spoil_multiplier`).